src/main/java/com/github/stefvanschie/inventoryframework/util/XMLUtil.java

```diff
@@ -23,34 +23,36 @@
      */
     @Nullable
     @Contract(pure = true)
-    public static Consumer<InventoryClickEvent> loadOnClickAttribute(@NotNull Object instance, @NotNull Element element,
-                                                                     @NotNull String name) {
+    public static <T extends Event> Consumer<T> loadOnEventAttribute(@NotNull Object instance, @NotNull Element element,
+                                                                     @NotNull Class<T> eventType, @NotNull String name) {
+        String attribute = element.getAttribute(name);
         for (Method method : instance.getClass().getMethods()) {
-            if (!method.getName().equals(element.getAttribute(name)))
+            if (!method.getName().equals(attribute))
                 continue;
 
             int parameterCount = method.getParameterCount();
-
+            boolean eventParameter;
             if (parameterCount == 0) {
-                return event -> {
-                    try {
-                        method.setAccessible(true);
-                        method.invoke(instance);
-                    } catch (IllegalAccessException | InvocationTargetException e) {
-                        e.printStackTrace();
-                    }
-                };
+                eventParameter = false;
             } else if (parameterCount == 1 &&
-                    InventoryClickEvent.class.isAssignableFrom(method.getParameterTypes()[0])) {
-                return event -> {
-                    try {
-                        method.setAccessible(true);
+                    eventType.isAssignableFrom(method.getParameterTypes()[0])) {
+                eventParameter = true;
+            } else {
+                continue;
+            }
+
+            return event -> {
+                try {
+                    method.setAccessible(true);
+                    if (eventParameter) {
                         method.invoke(instance, event);
-                    } catch (IllegalAccessException | InvocationTargetException e) {
-                        e.printStackTrace();
+                    } else {
+                        method.invoke(instance);
                     }
-                };
-            }
+                } catch (IllegalAccessException | InvocationTargetException e) {
+                    throw new RuntimeException("Error while invoking reflection-obtained event listener", e);
```

Comment:
Please change this exception type to `XMLReflectionException`.

Reply:
I'm dumm